src/hoc/withTheme.js

```diff
@@ -24,24 +24,32 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
 
     static contextTypes = {
       [CHANNEL]: PropTypes.func,
+      [CHANNEL_NEXT]: CONTEXT_CHANNEL_SHAPE,
     };
 
     state: { theme?: ?Object } = {};
-    unsubscribe: () => void;
+    unsubscribeId: number = -1
 
     componentWillMount() {
-      if (!this.context[CHANNEL]) {
-        throw new Error('[withTheme] Please use ThemeProvider to be able to use withTheme')
+      const styledContext = this.context[CHANNEL_NEXT]
+      if (!styledContext) {
```

Comment:
We could check for `!== undefined` and below in the unmount we could check for `unsubscribeId !== -1` instead? The latter doesn't make a big difference imho